resources/js/components/Room/MembersComponent.vue

```diff
@@ -276,18 +378,46 @@ export default {
       members: [], // list of all members
       createError: null, // error on adding new user as member
       editMember: null, // member to be edited
-      deleteMember: null, // member to be deleted
+      bulkEditRole: null, // role that will be applied to multiple members
+      removeMember: null, // member to be removed
       errors: {},
-      currentPage: 1
+      currentPage: 1,
+      selectedMembers: [],
+      displayedMembers: []
     };
   },
   methods: {
 
     /**
-     * Remove given member from member list
+     * select all checkbox toggled
+     * @param allSelected boolean select all rows
+     */
+    onAllRowsSelected (allSelected) {
+      if (!allSelected) {
+        // If checkbox is now unchecked, remove all selections
+        this.$refs.membersTable.clearSelected();
+      } else {
+        // If checkbox is now checked, select all rows
+        this.$refs.membersTable.selectAllRows();
+      }
+    },
+
+    /**
+     * Row selection changed
+     * @param selectedRows selected rows
      */
-    removeMemberItem: function (member) {
-      this.members.splice(this.members.findIndex(item => item.id === member.id), 1);
+    onRowsSelected (selectedRows) {
+      // set selected members to all selected table columns, exclude current user (prevent self edit/removed)
+      this.selectedMembers = selectedRows.filter(user => user.id !== this.currentUser.id);
+    },
+
+    /**
+     * Remove given members from member list, usable for both single removal or bulk removal
+     */
+    removeMemberItems: function (members) {
```

Comment:
This is the most efficient way to handle removal, but it might break the pagination. Maybe a reload is better.

Reply:
Resolve, because member list is client side pagination only